applications/leaderboard/Leaderboard/Program.cs

```diff
@@ -0,0 +1,525 @@
+                // Define create table statement for table #2.
+                createTableStatement =
+                @"CREATE TABLE Scores(
+                    PlayerId INT64 NOT NULL,
+                    Score INT64 NOT NULL,
+                    Timestamp TIMESTAMP NOT NULL OPTIONS(allow_commit_timestamp=true)
+                ) PRIMARY KEY(PlayerId, Timestamp), INTERLEAVE IN PARENT Players ON DELETE NO ACTION";
+                // Make the request.
+                cmd = connection.CreateDdlCommand(createTableStatement);
+                await cmd.ExecuteNonQueryAsync();
+            }
+        }
+
+        public static object InsertPlayers(string projectId,
+            string instanceId, string databaseId)
+        {
+            var responseTask =
+                InsertPlayersAsync(projectId, instanceId, databaseId);
+            Console.WriteLine("Waiting for operation to complete...");
+            responseTask.Wait();
+            Console.WriteLine($"Operation status: {responseTask.Status}");
+            Console.WriteLine($"Inserted records into sample database "
+                + $"{databaseId} on instance {instanceId}");
+            return ExitCode.Success;
+        }
+
+        public static async Task InsertPlayersAsync(string projectId,
+            string instanceId, string databaseId)
+        {
+            string connectionString =
+                $"Data Source=projects/{projectId}/instances/{instanceId}"
+                + $"/databases/{databaseId}";
+
+            // Get current max PlayerId.
+            Int64 maxPlayerId = 0;
+            using (var connection = new SpannerConnection(connectionString))
+            {
+                // Execute a SQL statement to get current MAX() of PlayerId.
+                var cmd = connection.CreateSelectCommand(
+                    @"SELECT MAX(PlayerId) as PlayerId FROM Players");
+                using (var reader = await cmd.ExecuteReaderAsync())
+                {
+                    while (await reader.ReadAsync())
+                    {
+                        long parsedValue;
+                        if (reader["PlayerId"] != DBNull.Value)
+                        {
+                            bool result = Int64.TryParse(
+                                reader.GetFieldValue<string>("PlayerId"), out parsedValue);
+                            if (result)
+                            {
+                                maxPlayerId = parsedValue;
+                            }
+                        }
+                    }
+                }
+            }
+
+            // Insert 100 player records into the Players table.
+            using (var connection = new SpannerConnection(connectionString))
+            {
+                await connection.OpenAsync();
+                using (var tx = await connection.BeginTransactionAsync())
+                using (var cmd = connection.CreateInsertCommand("Players", new SpannerParameterCollection
```

Comment:
To avoid collisions, getting the max playerid would have to happen in the same transaction.  But as my comment says above, I'd expect to see a GUID or the like.